modules/core/src/cuda/gpu_mat.cu

```diff
@@ -550,23 +550,30 @@ void cv::cuda::GpuMat::convertTo(OutputArray _dst, int rtype, Stream& stream) co
         return;
     }
 
-    CV_Assert( sdepth <= CV_64F && ddepth <= CV_64F );
+    CV_Assert( (sdepth <= CV_64F || (sdepth >= CV_64U && sdepth <= CV_32U)) &&
+               (ddepth <= CV_64F || (ddepth >= CV_64U && ddepth <= CV_32U)) );
```

Comment:
I propose to drop the check and add a check for the `funcs` table pointer. The condition will be simpler and we get simple source if truth.